persistence/src/main/scala/hmda/persistence/processing/SubmissionFSM.scala

```diff
@@ -226,10 +226,14 @@ class SubmissionFSM(submissionId: SubmissionId)(implicit val domainEventClassTag
     case Event(Sign, data) =>
       val status = hmda.model.fi.Signed
       updateStatus(status)
-      goto(Signed) applying SubmissionSigned(Submission(submissionId, status))
+      goto(Signed) applying SubmissionSigned(Submission(submissionId, status)) replying Some(status)
   }
 
   when(ValidatedWithErrors) {
+    case Event(Sign, data) =>
+      val status = hmda.model.fi.Signed
+      updateStatus(status)
+      goto(Signed) applying SubmissionSigned(Submission(submissionId, status)) replying Some(status)
```

Comment:
It looks like these 2 `replying Some(status)` can be moved under `when(Signed)`?